### ISSUE TITLE
azuread_group_role_management_policy throws 403 permission scope not granted error

### ISSUE THREAD

**Issue by cveld** [hashicorp/terraform-provider-azuread#1407]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* `azuread_group_role_management_policy`

### Terraform Configuration Files

```hcl

resource "azuread_group_role_management_policy" "example" {
  group_id = data.azuread_group.example.id
  role_id  = "member"
}

# created by SpokeBaseline/eu-dev
data "azuread_group" "example" {
  display_name = "sg-somegroup"
}
```

### Expected Behavior

The azuread provider should run fine without a permission error.

### Actual Behavior

```
╷
│ Error: Could not parse policy assignment ID, Could not list existing policy assignments, RoleManagementPolicyAssignmentClient.BaseClient.Get(): unexpected status 403 with OData error: UnknownError: {"errorCode":"PermissionScopeNotGranted","message":"Authorization failed due to missing permission scope RoleManagementPolicy.Read.AzureADGroup,RoleManagementPolicy.ReadWrite.AzureADGroup.","instanceAnnotations":[]}
│
│   with azuread_group_role_management_policy.example,
│   on main.tf line 2, in resource "azuread_group_role_management_policy" "example":
│    2: resource "azuread_group_role_management_policy" "example" {
│
│ Could not parse policy assignment ID, Could not list existing policy assignments,
│ RoleManagementPolicyAssignmentClient.BaseClient.Get(): unexpected status 403 with OData error: UnknownError:
│ {"errorCode":"PermissionScopeNotGranted","message":"Authorization failed due to missing permission scope
│ RoleManagementPolicy.Read.AzureADGroup,RoleManagementPolicy.ReadWrite.AzureADGroup.","instanceAnnotations":[]}
╵
```

### Steps to Reproduce

1. Running as user
2. User has global administrator role in Entra
1. `terraform apply`

### References

* #26471
* https://github.com/Azure/azure-cli/issues/22775
* https://github.com/Azure/azure-cli/issues/28854

**Comment by manicminer** [comment]
Hi @cveld, thanks for reporting this. As you have suggested, this does seem to be an issue with the delegated permissions obtained by Azure CLI. At this time there is little we can do about it, except suggest, as it has been in the [linked issue](https://github.com/Azure/azure-cli/issues/22775), that you use application credentials to authenticate instead of a user account (and furthermore, our recommendation would be to use the provider's native support rather than using Azure CLI). I recognise this may be undesirable or infeasible due to organizational policies or other concerns, however we are unfortunately blocked on Azure CLI at this time.

**Comment by cveld** [comment]
@manicminer if user identity is not supported as this moment, can you can add that to the docs? the current version gave the impression to me that it should work:

> When authenticated with a user principal, this resource requires Global Administrator directory role, or the Privileged Role Administrator role in Identity Governance.

**Comment by manicminer** [comment]
Yes, I'm hoping soon to audit the documented scopes for all resources and update accordingly.

**Comment by cveld** [comment]
Maybe azuread could implement its own user auth and provide a customizable client id 😅 well I am more a fan of delegating access tokens to az cli. But never hit into the limitation of the client id permissions before. That surprised me.

**Comment by Poltergeisen** [comment]
any update on this? I'm hitting this error when doing a tf plan

**Comment by ronmanthe** [comment]
Any update? I am running into this issue as well on hashicorp/azuread 3.3.0

**Comment by Poltergeisen** [comment]
has anyone found a workaround for this? We've been having this issue for a year or so with no great way to solve it for local terraform plans.

**Comment by Sebastian-Materna** [comment]
Faced also this issue. I'm using a Service Principle with admin consent for both permissions:
- RoleManagementPolicy.Read.AzureADGroup
- RoleManagementPolicy.ReadWrite.AzureADGroup

It's weird I know, but after granting the consent and waiting some minutes I was able to plan and apply again. Maybe this also helps you too.
